node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+    use itertools::Either;
+    use masq_lib::logger::Logger;
+    use masq_lib::test_utils::logging::{init_test_logging, TestLogHandler};
+    use masq_lib::utils::convert_collection;
+    use std::collections::HashMap;
+    use std::panic::{catch_unwind, AssertUnwindSafe};
+    use std::sync::{Arc, Mutex};
+    use std::time::{Duration, SystemTime};
+    use std::{usize, vec};
+    use thousands::Separable;
+    use web3::types::U256;
+
+    #[test]
+    #[should_panic(expected = "Broken code: Called the null implementation of \
+        the unallocated_cw_service_fee_balance_minor() method in PaymentAdjusterInner")]
+    fn payment_adjuster_new_is_created_with_inner_null() {
+        let result = PaymentAdjusterReal::new();
+
+        let _ = result.inner.unallocated_cw_service_fee_balance_minor();
+    }
+
+    fn test_initialize_inner_works(
+        required_adjustment: Adjustment,
+        expected_tx_fee_limit_opt_result: Option<u16>,
+    ) {
+        let mut subject = PaymentAdjusterReal::default();
+        let cw_service_fee_balance = 111_222_333_444;
+        let largest_exceeding_balance_recently_qualified = 3_555_666;
+        let now = SystemTime::now();
+
+        subject.initialize_inner(
+            cw_service_fee_balance,
+            required_adjustment,
+            largest_exceeding_balance_recently_qualified,
+            now,
+        );
+
+        assert_eq!(subject.inner.now(), now);
+        assert_eq!(
+            subject.inner.transaction_fee_count_limit_opt(),
+            expected_tx_fee_limit_opt_result
+        );
+        assert_eq!(
+            subject.inner.original_cw_service_fee_balance_minor(),
+            cw_service_fee_balance
+        );
+        assert_eq!(
+            subject.inner.unallocated_cw_service_fee_balance_minor(),
+            cw_service_fee_balance
+        );
+        assert_eq!(
+            subject.inner.largest_exceeding_balance_recently_qualified(),
+            largest_exceeding_balance_recently_qualified
+        )
+    }
+
+    #[test]
+    fn initialize_inner_processes_works() {
+        test_initialize_inner_works(Adjustment::ByServiceFee, None);
+        test_initialize_inner_works(
+            Adjustment::TransactionFeeInPriority {
+                affordable_transaction_count: 5,
+            },
+            Some(5),
+        );
+    }
+
+    #[test]
+    fn search_for_indispensable_adjustment_happy_path() {
+        init_test_logging();
+        let test_name = "search_for_indispensable_adjustment_gives_negative_answer";
+        let mut subject = PaymentAdjusterReal::new();
+        subject.logger = Logger::new(test_name);
+        // Service fee balance > payments
+        let input_1 = make_input_for_initial_check_tests(
+            Some(TestConfigForServiceFeeBalances {
+                account_balances: Either::Right(vec![
+                    gwei_to_wei::<u128, u64>(85),
+                    gwei_to_wei::<u128, u64>(15) - 1,
```

Comment:
The `u128` seems unnecessary here. There's a suggestion to use the `_u64` with the numbers that are provided as arguments.